content/en/blog/_posts/2021-XX-XX-writing-a-controller-for-pod-labels.md

```diff
@@ -0,0 +1,473 @@
+cluster.
+
+Let's keep the operator running and, in another terminal, create a new Pod:
+
+```bash
+kubectl run --image=nginx my-nginx
+```
+
+The operator should quickly print some logs, indicating that it reacted to the
+Pod's creation and subsequent changes in status:
+
+```text
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+```
+
+Lets check the Pod's labels:
+
+```terminal
+$ kubectl get pod my-nginx --show-labels
+NAME       READY   STATUS    RESTARTS   AGE   LABELS
+my-nginx   1/1     Running   0          11m   run=my-nginx
+```
+
+Let's add an annotation to the Pod so that our controller knows to add our
+dynamic label to it:
+
+```bash
+kubectl annotate pod my-nginx padok.fr/add-pod-name-label=true
+```
+
+Notice that the controller immediately reacted and produced a new line in its
+logs:
+
+```text
+INFO    controllers.Pod adding label    {"pod": "default/my-nginx"}
+```
+
+```terminal
+$ kubectl get pod my-nginx --show-labels
+NAME       READY   STATUS    RESTARTS   AGE   LABELS
+my-nginx   1/1     Running   0          13m   padok.fr/pod-name=my-nginx,run=my-nginx
+```
+
+Bravo! You just successfully wrote a Kubernetes controller capable of adding
+labels with dynamic values to resources in your cluster.
+
+Controller and, more generally, operators both big and small, can be an
+important part of your Kubernetes journey. Writing operators is easier now than
+it has ever been. The possibilities are endless.
+
+## What next?
+
+If you want to go further, I recommend starting by deploying your controller or
+operator inside a cluster. The `Makefile` generated by the Operator SDK will do
+most of the work.
+
+When deploying an operator to production, it is always a good idea to implement
+robust testing. The first step in that direction is to write unit tests.
+[This documentation][operator-sdk-testing] will guide you in writing tests for
+your operator. I wrote tests for the operator we just wrote; you can find all of
+my code in [this GitHub repository][github-repo].
+
+## How to learn more?
+
+The [Operator SDK documentation][operator-sdk-docs] goes into detail on how you
+can go further and implement more complex operators.
+
+When modeling a more complex use-case, a single controller acting on built-in
+Kubernetes types may not be enough. You may need to build a fully fledged
+operator with [Custom Resource Definitions (CRDs)][custom-resource-definitions]
+and multiple controllers. The Operator SDK is a great tool to help you do this.
+
+If you want to discuss building an operator, hit me up on Twitter [@ArthurBusser][twitter-profile]!
+
+---
+
+_[Arthur Busser][twitter-profile] is a site reliability engineer at
+[Padok][padok] where he works on tooling and automation._
```

Comment:
In the Kubernetes blog, we rarely include this kind of final paragraph about the author at all. Interested parties can search the web for “Arthur Busser Padok” and learn more if they want to.

Reply:
@busser, the blog post's byline will display your name and organization at the top of the article. Would you mind removing these two lines?

Reply:
Done 👍